pkg/loader/compose/compose.go

```diff
@@ -29,12 +30,107 @@ import (
 	"github.com/docker/libcompose/config"
 	"github.com/docker/libcompose/lookup"
 	"github.com/docker/libcompose/project"
+	"github.com/fatih/structs"
 	"github.com/kubernetes-incubator/kompose/pkg/kobject"
 )
 
 type Compose struct {
 }
 
+// checkUnsupportedKey checks if libcompose project contains
+// keys that are not supported by this loader.
+// list of all unsupported keys are stored in unsupportedKey variable
+// returns list of unsupported YAML keys from docker-compose
+func checkUnsupportedKey(composeProject *project.Project) []string {
+
+	// list of all unsupported keys for this loader
+	// this is map to make searching for keys easier
+	// also counts how many times was given key found in service
+	// to make sure that we show warning only once for every key
+	var unsupportedKey = map[string]int{
+		"CgroupParent":  0,
+		"Devices":       0,
+		"DependsOn":     0,
+		"DNS":           0,
+		"DNSSearch":     0,
+		"DomainName":    0,
+		"EnvFile":       0,
+		"Extends":       0,
+		"ExternalLinks": 0,
+		"ExtraHosts":    0,
+		"Hostname":      0,
+		"Ipc":           0,
+		"Logging":       0,
+		"MacAddress":    0,
+		"MemLimit":      0,
+		"MemSwapLimit":  0,
+		"NetworkMode":   0,
+		"Pid":           0,
+		"SecurityOpt":   0,
+		"ShmSize":       0,
+		"StopSignal":    0,
+		"VolumeDriver":  0,
+		"Uts":           0,
+		"ReadOnly":      0,
+		"StdinOpen":     0,
+		"Tty":           0,
+		"Ulimits":       0,
+		"Dockerfile":    0,
+		"Net":           0,
+		"Networks":      0, // there are special checks for Network in checkUnsupportedKey function996607
+	}
+
+	// collect all keys found in project
+	var keysFound []string
+
+	// Root level keys
+	// volume config and network config are not supported
+	if len(composeProject.NetworkConfigs) > 0 {
+		keysFound = append(keysFound, "root level networks")
+	}
+	if len(composeProject.VolumeConfigs) > 0 {
+		keysFound = append(keysFound, "root level volumes")
+	}
+
+	for _, serviceConfig := range composeProject.ServiceConfigs.All() {
+		// this reflection is used in check for empty arrays
+		val := reflect.ValueOf(serviceConfig).Elem()
+		s := structs.New(serviceConfig)
+
+		for _, f := range s.Fields() {
+			if f.IsExported() && !f.IsZero() {
+				// IsZero returns false for empty array/slice ([])
+				// this check if field is Slice, and then it checks its size
+				if field := val.FieldByName(f.Name()); field.Kind() == reflect.Slice {
+					if field.Len() == 0 {
+						// array is empty it doesn't metter if it is in unsupportedKey or not
+						continue
+					}
+				}
+
+				if counter, ok := unsupportedKey[f.Name()]; ok {
```

Comment:
Technically It could be moved there, I kept it in the same order as it was before.

I wouldn't say a lot of processing as happening before. It is all just simple checks. Plus it is not like there will ever be hundreds of keys.
But I updated it anyway ;-)  Plus I changed int counter in unsuportedKeys to bool. 
Because all we need is to know if there already was that key or not ;-)